firestore-bigquery-export/scripts/gen-schema-view/src/schema.ts

```diff
@@ -100,76 +128,110 @@ export class FirestoreBigQuerySchemaViewFactory {
     let latestView = dataset.table(latestSchemaViewName);
     const [latestViewExists] = await latestView.exists();
 
+    let result = userSchemaView(
+      datasetId,
+      rawChangeLogTableName,
+      firestoreSchema
+    );
+    let bigQueryFields = result.fields;
+    const options = {
+      friendlyName: changeLogSchemaViewName,
+      view: result.viewInfo,
+    };
     if (!viewExists) {
-      const schemaView = userSchemaView(
-        datasetId,
-        rawChangeLogTableName,
-        firestoreSchema
-      );
       logs.bigQuerySchemaViewCreating(
         changeLogSchemaViewName,
         firestoreSchema,
-        schemaView.query
+        result.viewInfo.query
       );
-      const options = {
-        friendlyName: changeLogSchemaViewName,
-        view: schemaView,
-      };
       await view.create(options);
       logs.bigQuerySchemaViewCreated(changeLogSchemaViewName);
     }
+    await view.setMetadata({
+      schema: decorateSchemaWithChangelogFields({
+        fields: bigQueryFields,
+      }),
+    });
 
+    result = latestConsistentSnapshotSchemaView(
+      datasetId,
+      latestRawViewName,
+      firestoreSchema
+    );
+    bigQueryFields = result.fields;
+    const latestOptions = {
+      fiendlyName: latestSchemaViewName,
+      view: result.viewInfo,
+    };
     if (!latestViewExists) {
-      const latestSchemaView = buildSchemaViewQuery(
-        datasetId,
-        latestRawViewName,
-        firestoreSchema
-      );
       logs.bigQuerySchemaViewCreating(
         latestSchemaViewName,
         firestoreSchema,
-        latestSchemaView
+        result.viewInfo.query
       );
-      const latestOptions = {
-        fiendlyName: latestSchemaViewName,
-        view: latestSchemaView,
-      };
       await latestView.create(latestOptions);
       logs.bigQueryViewCreated(latestSchemaViewName);
     }
+    await latestView.setMetadata({
+      schema: decorateSchemaWithChangelogFields({
+        fields: bigQueryFields,
+      }),
+    });
 
     return view;
   }
 }
 
 /**
- * A wrapper around `buildSchemaView` that can be passed into BigQuery's
- * `table.create`.
+ * Given a BigQuery schema returned from either `userSchemaView` or
+ * `latestConsistentSnapshotSchemaView`, inherit the appropriate
+ * fields from the raw changelog schema and return the combined schemas.
+ */
+function decorateSchemaWithChangelogFields(schema: any): any {
+  let decorated: any = { fields: schema.fields };
+  const changelogSchemaFields: any[] = RawChangelogSchema.fields;
+  for (let i = 0; i < changelogSchemaFields.length; i++) {
+    if (
+      changelogSchemaFields[i].name == "event_id" ||
+      changelogSchemaFields[i].name == "data"
+    ) {
+      continue;
+    }
+    changelogSchemaFields[i].mode = "NULLABLE";
```

Comment:
This seems to be modifying `RawChangelogSchema.fields`.
Is this intentional? Try to make this method side-effect free.